### PR TITLE
Add a C&U-based alert test

### PR DESCRIPTION
**Pull request by mfalesni**
{{pytest: 'cfme/tests/control/test_alerts.py -v --long-running'}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 15a3243a023da16b90daedb0920df7cd2d7bdfde:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 6a54a828f51b45b24ee795bed9cbea44c196ea9a:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 820290bb664759cd322a5c13d2464b25d88b648f:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit ffa58478c91c2ef38425f2320e8f41f4a496d488:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 111fd3fd42b6c71b630913bf17da44894a7e49ed:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 64f9594d93ab23a01cda69b4aab6660472f8955a:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit a8c9be993a0baa87e42409b650ec3ae60b7e5819:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 1474bd88e3b9e1d8bbfd038de001f2ce5b4b2627:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 3559ece5c60889f1e9d99490c1d7210fe76080b6:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit a2163710a0ced874cf39be3ab933f997fe703eff:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 4458774edccb18d781791c2828ff616b737cdde0:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit c94a1d1e5ee8f59fd95c7e51534b15004227b091:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by mfalesni** [comment]
Automates https://bugzilla.redhat.com/show_bug.cgi?id=1231889, https://bugzilla.redhat.com/show_bug.cgi?id=1232022

**Comment by dajoRH** [comment]
Lint report for commit 2a0976079d3815cbbd4f1fed9401eead6e185742:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 7fccf6ddef3e5bde35c0a2bbc8837e1cb932bad5:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 963556c15b76a04bbcacfbcb449fd577f9a8e675:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit cbdcb3be673f853e598720cd530478c8ed00f5d7:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit eced8cdb41e195462426d26c7682ca46d14724ed:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 5db92d920dbdee4096e8dcb1cd8286f799df96b9:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 89e172be5b615afe8c9ae491b1cba138fc416fed:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 94abd9ddae7a35c87ebeb4c4ef54136c05f76b91:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit fa3f2c63cd909ce3ca48c807dea858d766ae1651:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 64d4f63bfe431b676561284606575e8cf304b1fc:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 05639a7ce11235c4eb7dee7ad527d01a9b4ee62f:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 06fbb1a220ba84dee9f241d98663b3afbbaa41a7:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 487e6d4b0d6f5c52c5992c93ac528b7a288904ac:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit fe01ff31e66ebf87266935bf435bfeaf3da074f2:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit a6e5a0a8bc23e4dd30e406cb2f6b7b5086d685d2:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/tests/control/test_alerts.py:14:1: F401 'clear_providers' imported but unused
```
:trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 4589348edcec55928c1244b5aec2f88a4d4068f7:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 08add9b3e381874500bafb76bbe74c0f269c7fce:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit f174d8ae9baa62e7b5c03fbe8d5d5eb9f36ee61e:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 65e6dbc0c2b4da0012c1ddcca9fb1eb87fd09c55:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 0a9f0b36b32b0d912dd3564e64b56426dfedf134:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 8bd9d7c63b31fb503548e9887a40f4e0ed8272cb:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 3a08124a76e1a1e3170d7f5922ad6a962a1bedd8:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit a8905d6319ab31f0ef58dc606f9f5942600a957d:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 4480467f3aedcfd16820e5fe8dc25b41875f8f87:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 3a40d533ef8d532d7c4997cf5147a3cbaf071673:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 955d8209f1dbd4afadeaadd77187c1ce1c2f9cca:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 3ef4e5c80473d617c4371ce60c0c0f46f04ac5b7:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit c2bb85ac6ed7ba73d67a68d904fb57b09b1b6b39:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by mfalesni** [comment]
@psav So, the last run passed all green, so I squashed the commits. The SNMP test is made skipping because something was hogging it, will resolve that later.

**Comment by mfalesni** [comment]
@psav @seandst So, although it failed, last time it went 100% green. The difference is none, just squashing. Asking for review.

**Comment by dajoRH** [comment]
Lint report for commit 27c6c7c08fc0e013358a7aaa1889765671dfc634:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 87cb27354aea917652614afba589d4bd02ae7a32:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by mfalesni** [comment]
WAiting for VM Unify to go in for rebase.

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit bf5ee57feea746348097bcb6b6c6c88791437918:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by psav** [comment]
Shuld the timelines stuff go into test_timelines?

**Comment by dajoRH** [comment]
Lint report for commit d226292be47f028b97841f7bec550084957c45ab:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by mfalesni** [comment]
@psav Not sure. This tests that the alert creates the timeline event. I think because of the fixture support in this test module it should stay here.

**Comment by dajoRH** [comment]
Lint report for commit 6034cd4c895bb425b61a8f5c1bcd974a0ed9b7a2:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 2b60703be0a71357c624376ea19dc7fd12d99ea2:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit b60a22ac3240e9aeddfab7cc4b64f95d1e8ad063:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit ae180984efca941a67a495439a87c70b2be7a5d7:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by psav** [comment]
I agree!

**Comment by dajoRH** [comment]
Lint report for commit f277962a8cc26a8262a6fb8c2bac1e27b80c39ae:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajohnso** [comment]
@mfalesni , so where are we at with this one, I'll execute a retest but it would be good to determine if this is a application failure in 5.5, we block it, and then merge kind of thing.

**Comment by dajohnso** [comment]
Dude, I did not mean to close, I guess I was not reading the buttons ... whoopsie

**Comment by mfalesni** [comment]
@dajohnso It looks like it does not like the e-mail form first.

**Comment by dajoRH** [comment]
Lint report for commit bb2ae48f0487e65272aaaba70022be58f6584ba8:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 6256be41bd3b8279c41e2d32e968fad7c0a004d4:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit a62165223476f7697039441636edb4eaff49a39f:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by mfalesni** [comment]
@psav This looks good now. Can we mergy?

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 2071a565bd5ce33d3ca0c055eb90cb8f3e5d01e0:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

File: cfme/common/vm.py [hunk 1]
```diff
@@ -346,6 +347,10 @@ def open_edit(self):
         self.load_details(refresh=True)
         cfg_btn(self.TO_OPEN_EDIT)
 
+    def open_timelines(self):
+        self.load_details(refresh=True)
+        mon_btn("Timelines")
```
Comment (psav) [inline cfme/common/vm.py]: So glad that the interim toolbar fix means we don't have to change anything :)

File: cfme/tests/control/test_alerts.py [hunk 2]
```diff
@@ -121,25 +192,182 @@ def test_alert_vm_turned_on_more_than_twice_in_past_15_minutes(
...
+            if info.get("Event Source") != "MiqAlert":
+                continue
+            if info["Source VM"] == vm_name:
+                return True
+        return False
+
+    wait_for(_timeline_event_present, num_sec=30 * 60, delay=15, message="timeline event present")
+
+
+@pytest.mark.skipif(True, reason="SNMP hogs the tests.")
```
Comment (psav) [inline cfme/tests/control/test_alerts.py]: Is it intentional to always skip this? If it is, we should uncollect it instead.

Comment (mfalesni) [inline cfme/tests/control/test_alerts.py]: @psav  SNMP installation hogged the tests so I left it for later to be resolved.

File: cfme/web_ui/__init__.py [hunk 3]
```diff
@@ -3423,3 +3344,84 @@ def clear(self):
...
+                continue
+            if email in self.user_emails:
+                sel.select(self.fields.user_emails, sel.ByValue(email))
+            else:
+                fill(self.fields.manual_input, email)
+                sel.click(self.fields.add_email_manually)
+                assert email in self.to_emails, "Adding e-mail '%s' manually failed!" % email
+
+
+@fill.method((EmailSelectForm, basestring))
```
Comment (psav) [inline cfme/web_ui/__init__.py]: would ```object``` be better than having all the registered types?

File: utils/providers.py [hunk 4]
```diff
@@ -267,6 +267,24 @@ def is_provider_setup(provider_key):
     return get_crud(provider_key).exists
 
 
+def existing_providers():
+    """Lists all providers that are already set up in the appliance."""
+    return filter(is_provider_setup, list_all_providers())
```
Comment (psav) [inline utils/providers.py]: Without looking, just presuming this uses db?

File: cfme/web_ui/__init__.py (hunk 3 above)
Comment (mfalesni) [inline cfme/web_ui/__init__.py]: @psav Possibly, but by stating it here you can clearly see what can you pass to the fill method.

File: utils/providers.py [hunk 5]
```diff
@@ -267,6 +267,24 @@ def is_provider_setup(provider_key):
...
+
+
+def get_provider(provider_key):
+    """Returns a CRUD object of the provider key.
```
Comment (psav) [inline utils/providers.py]: This is not really accurate. It doesn't just return the crud, it sets it up in the appliance.

File: utils/providers.py (hunk 4 above)
Comment (mfalesni) [inline utils/providers.py]: Yes, it does. It uses the crud.exists property that looks in the database.

File: utils/providers.py (hunk 5 above)
Comment (mfalesni) [inline utils/providers.py]: This is a remainder of the older code. I think I will use the get_crud instead of this one.

**Comment by mfalesni** [comment]
@psav I removed the get_provider and replaced it with using get_crud so I think it might be merged now.

**Comment by dajoRH** [comment]
Lint report for commit 36a3c0e15dc522c470fe306cb78e4f873701241b:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajohnso** [comment]
@mfalesni @psav Can we merge this?

**Comment by mfalesni** [comment]
@dajohnso I am waiting for psav to review and merge. The tests themselves work.

**Comment by dajoRH** [comment]
Lint report for commit 3bc2bcd70cffb773d1fe2b5967f0557a1d220565:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 8e72f535bc3234283a16250b9f6fd462a3c01d81:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Pull request merged**